subsys/bluetooth/host/scan.c

```diff
@@ -507,50 +551,154 @@ static uint8_t get_adv_type(uint8_t evt_type)
 	}
 }
 
+/* Convert extended adv report evt_type field to adv props */
+static uint16_t get_adv_props_extended(uint16_t evt_type)
+{
+	/* Converts from BT_HCI_LE_ADV_EVT_TYPE_* to BT_GAP_ADV_PROP_*
+	 * The first 4 bits are the same (conn, scan, direct, scan_rsp).
+	 * Bit 4 must be flipped as the meaning of 1 is opposite (legacy -> extended)
+	 * The rest of the bits are zeroed out.
+	 */
+	return (evt_type ^ BT_HCI_LE_ADV_EVT_TYPE_LEGACY) & BIT_MASK(5);
+}
```

Comment:
I moved this check to a separate function with a comment to make the conversion clearer. The logic from before seems correct, but was a bit hard to decode. @alwa-nordic Had some questions about this.